lib/mfg.c

```diff
@@ -216,10 +225,18 @@ static int secure_config_get(struct switchtec_dev *dev,
 	state->jtag_bl1_unlock_allowed = !!(reply.cfg & 0x0100);
 	state->jtag_post_bl1_unlock_allowed = !!(reply.cfg & 0x0200);
 
-	state->spi_clk_rate =
-		(reply.cfg >> SWITCHTEC_CLK_RATE_BITSHIFT) & 0x0f;
-	if (state->spi_clk_rate == 0)
-		state->spi_clk_rate = SWITCHTEC_SPI_RATE_25M;
+	spi_clk = (reply.cfg >> SWITCHTEC_CLK_RATE_BITSHIFT) & 0x0f;
+	if (spi_clk == 0) {
+		if (switchtec_gen(dev) == SWITCHTEC_GEN5)
```

Comment:
I probably would have seperated this into two patches: one making the float change, the second adding GEN5 support.